### PR TITLE
[TS migration] Migrate MoneyTemporaryForRefactorRequestParticipantsSe…

### PR DESCRIPTION
**Pull request by pasyukevich**
…lector.js to MoneyRequestParticipantsSelector.tsx

### Fixed Issues

$ https://github.com/Expensify/App/issues/29107 - MoneyTemporaryForRefactorRequestParticipantsSelector.js to MoneyRequestParticipantsSelector.tsx migration part
PROPOSAL:

### Tests

1. Tap on the green plus button (FAB)
2. Select Request Money/Manual
3. Select any currency and amount
4. Press "Next" button
5. Search for a needed participant
6. Select it 
7. Verify that selector worked as expected 

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

[android-native-converted.webm](https://github.com/Expensify/App/assets/23215729/3bbefd75-b7e1-4e1a-afbf-71acaead28b9)

</details>

<details>
<summary>Android: mWeb Chrome</summary>

[android-web-converted.webm](https://github.com/Expensify/App/assets/23215729/32f01a43-bd1b-4477-9616-9ba0ae257cf7)

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/23215729/a3d0713e-300a-465d-bca0-32f60ac8575e

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/23215729/85dad674-3a69-42d0-b409-82171f86a6ae

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/23215729/77fcb4bf-8ae1-4e07-8b99-5914d46d572d

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/23215729/6c94e8b1-a739-4ff8-8a08-217ef24c607c

</details>

### CODE REVIEW

File: src/pages/iou/request/MoneyRequestParticipantsSelector.tsx [hunk 1]
```diff
@@ -269,29 +265,29 @@ function MoneyTemporaryForRefactorRequestParticipantsSelector({participants, onF
     const headerMessage = useMemo(
         () =>
             OptionsListUtils.getHeaderMessage(
-                lodashGet(newChatOptions, 'personalDetails', []).length + lodashGet(newChatOptions, 'recentReports', []).length !== 0,
-                Boolean(newChatOptions.userToInvite),
+                ((newChatOptions as Options)?.personalDetails ?? []).length + ((newChatOptions as Options)?.recentReports ?? []).length !== 0,
+                Boolean((newChatOptions as Options)?.userToInvite),
```
Comment (pac-guerreiro) [inline src/pages/iou/request/MoneyRequestParticipantsSelector.tsx:269]:
```suggestion
                !!(newChatOptions as Options)?.userToInvite,
```

**Comment by DylanDylann** [comment]
@pasyukevich Please speed up this one, this is the last issue of [this epic](https://github.com/Expensify/App/issues/29107) that is over the deadline (1 May)

File: src/pages/iou/request/MoneyRequestParticipantsSelector.tsx (hunk 1 above)
Comment (pasyukevich) [reply src/pages/iou/request/MoneyRequestParticipantsSelector.tsx:255]: I think it better to leave straight-type conversion here since it is less complicated to understand

File: src/pages/iou/request/MoneyRequestParticipantsSelector.tsx [hunk 2]
```diff
@@ -79,10 +72,10 @@ function MoneyTemporaryForRefactorRequestParticipantsSelector({participants, onF
         shouldInitialize: didScreenTransitionEnd,
     });
 
-    const offlineMessage = isOffline ? [`${translate('common.youAppearToBeOffline')} ${translate('search.resultsAreLimited')}`, {isTranslated: true}] : '';
+    const offlineMessage: MaybePhraseKey = isOffline ? [`${translate('common.youAppearToBeOffline')} ${translate('search.resultsAreLimited')}`, {isTranslated: true}] : '';
 
     const isIOUSplit = iouType === CONST.IOU.TYPE.SPLIT;
-    const isCategorizeOrShareAction = [CONST.IOU.ACTION.CATEGORIZE, CONST.IOU.ACTION.SHARE].includes(action);
+    const isCategorizeOrShareAction = ([CONST.IOU.ACTION.CATEGORIZE, CONST.IOU.ACTION.SHARE] as string[]).includes(action);
```
Comment (kubabutkiewicz) [inline src/pages/iou/request/MoneyRequestParticipantsSelector.tsx:78]: nab: you can use `some` instead of `includes` to avoid assertion

File: src/pages/iou/request/MoneyRequestParticipantsSelector.tsx [hunk 3]
```diff
@@ -112,14 +105,14 @@ function MoneyTemporaryForRefactorRequestParticipantsSelector({participants, onF
             // sees the option to submit an expense from their admin on their own Workspace Chat.
             (iouType === CONST.IOU.TYPE.SUBMIT || iouType === CONST.IOU.TYPE.SPLIT) && action !== CONST.IOU.ACTION.SUBMIT,
 
-            (canUseP2PDistanceRequests || iouRequestType !== CONST.IOU.REQUEST_TYPE.DISTANCE) && !isCategorizeOrShareAction,
+            (canUseP2PDistanceRequests ?? iouRequestType !== CONST.IOU.REQUEST_TYPE.DISTANCE) && !isCategorizeOrShareAction,
```
Comment (kubabutkiewicz) [inline src/pages/iou/request/MoneyRequestParticipantsSelector.tsx:108]: are you sure this is giving same results?

File: src/pages/iou/request/MoneyRequestParticipantsSelector.tsx [hunk 4]
```diff
@@ -112,14 +105,14 @@ function MoneyTemporaryForRefactorRequestParticipantsSelector({participants, onF
...
             false,
             {},
             [],
             false,
             {},
             [],
-            (canUseP2PDistanceRequests || iouRequestType !== CONST.IOU.REQUEST_TYPE.DISTANCE) && !isCategorizeOrShareAction,
+            (canUseP2PDistanceRequests ?? iouRequestType !== CONST.IOU.REQUEST_TYPE.DISTANCE) && !isCategorizeOrShareAction,
```
Comment (kubabutkiewicz) [inline src/pages/iou/request/MoneyRequestParticipantsSelector.tsx:115]: same here

File: src/pages/iou/request/MoneyRequestParticipantsSelector.tsx [hunk 5]
```diff
@@ -131,7 +124,7 @@ function MoneyTemporaryForRefactorRequestParticipantsSelector({participants, onF
             isCategorizeOrShareAction ? 0 : undefined,
         );
 
-        const formatResults = OptionsListUtils.formatSectionsFromSearchTerm(debouncedSearchTerm, participants, chatOptions.recentReports, chatOptions.personalDetails, personalDetails, true);
+        const formatResults = OptionsListUtils.formatSectionsFromSearchTerm(debouncedSearchTerm, participants.map((participant) => ({...participant, reportID: participant.reportID ?? ''})), chatOptions.recentReports, chatOptions.personalDetails, personalDetails, true);
```
Comment (kubabutkiewicz) [inline src/pages/iou/request/MoneyRequestParticipantsSelector.tsx:127]: why this map is added ?

File: src/pages/iou/request/MoneyRequestParticipantsSelector.tsx [hunk 6]
```diff
@@ -255,28 +251,28 @@ function MoneyTemporaryForRefactorRequestParticipantsSelector({participants, onF
...
+    const hasPolicyExpenseChatParticipant = participants.some((participant) => participant.isPolicyExpenseChat);
     const shouldShowSplitBillErrorMessage = participants.length > 1 && hasPolicyExpenseChatParticipant;
 
     // canUseP2PDistanceRequests is true if the iouType is track expense, but we don't want to allow splitting distance with track expense yet
     const isAllowedToSplit =
-        (canUseP2PDistanceRequests || iouRequestType !== CONST.IOU.REQUEST_TYPE.DISTANCE) &&
-        ![CONST.IOU.TYPE.PAY, CONST.IOU.TYPE.TRACK, CONST.IOU.TYPE.INVOICE].includes(iouType) &&
-        ![CONST.IOU.ACTION.SHARE, CONST.IOU.ACTION.SUBMIT, CONST.IOU.ACTION.CATEGORIZE].includes(action);
+        (canUseP2PDistanceRequests ?? iouRequestType !== CONST.IOU.REQUEST_TYPE.DISTANCE) &&
+        !([CONST.IOU.TYPE.PAY, CONST.IOU.TYPE.TRACK, CONST.IOU.TYPE.INVOICE] as string[]).includes(iouType) &&
```
Comment (kubabutkiewicz) [inline src/pages/iou/request/MoneyRequestParticipantsSelector.tsx:271]: nab: same here you can use `some` instead of `includes`

File: src/pages/iou/request/MoneyRequestParticipantsSelector.tsx (hunk 5 above)
Comment (pasyukevich) [reply src/pages/iou/request/MoneyRequestParticipantsSelector.tsx:127]: Because of type diff - Participant[] vs ReportUtils.OptionData[] 

ReportUtils.OptionData[] requires reportID with value

File: src/pages/iou/request/MoneyRequestParticipantsSelector.tsx (hunk 6 above)
Comment (pasyukevich) [reply src/pages/iou/request/MoneyRequestParticipantsSelector.tsx]: updated

File: src/pages/iou/request/MoneyRequestParticipantsSelector.tsx (hunk 4 above)
Comment (pasyukevich) [reply src/pages/iou/request/MoneyRequestParticipantsSelector.tsx]: updated

File: src/pages/iou/request/MoneyRequestParticipantsSelector.tsx (hunk 3 above)
Comment (pasyukevich) [reply src/pages/iou/request/MoneyRequestParticipantsSelector.tsx]: reverted since it can produce bugs in case of a boolean value

**Review by kubabutkiewicz — APPROVED** [review]

File: src/pages/iou/request/MoneyRequestParticipantsSelector.tsx [hunk 7]
```diff
@@ -192,10 +197,10 @@ function MoneyTemporaryForRefactorRequestParticipantsSelector({participants, onF
             ];
 
             if (iouType === CONST.IOU.TYPE.INVOICE) {
-                const primaryPolicy = Policy.getPrimaryPolicy(activePolicyID);
+                const primaryPolicy = Policy.getPrimaryPolicy(activePolicyID ?? undefined);
```
Comment (blazejkustra) [inline src/pages/iou/request/MoneyRequestParticipantsSelector.tsx:200]: You can revert this change and instead adjust getPrimaryPolicy function:

```ts
function getPrimaryPolicy(activePolicyID?: OnyxEntry<string>): Policy | undefined {
    const activeAdminWorkspaces = PolicyUtils.getActiveAdminWorkspaces(allPolicies);
    const primaryPolicy = allPolicies?.[activePolicyID ?? ''];

    return primaryPolicy ?? activeAdminWorkspaces[0];
}
```

```suggestion
                const primaryPolicy = Policy.getPrimaryPolicy(activePolicyID);
```

File: src/pages/iou/request/MoneyRequestParticipantsSelector.tsx [hunk 8]
```diff
@@ -23,46 +19,43 @@ import usePermissions from '@hooks/usePermissions';
...
+    iouRequestType: IOURequestType;
 
     /** The action of the IOU, i.e. create, split, move */
-    action: PropTypes.oneOf(lodashValues(CONST.IOU.ACTION)),
-};
-
-const defaultProps = {
-    participants: [],
-    action: CONST.IOU.ACTION.CREATE,
+    action?: IOUAction;
```
Comment (blazejkustra) [inline src/pages/iou/request/MoneyRequestParticipantsSelector.tsx:48]: Looking at the code this can be a required prop

**Review by blazejkustra — APPROVED** [review]
Minor comments, other than that LGTM!

**Comment by DylanDylann** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

File: src/pages/iou/request/MoneyRequestParticipantsSelector.tsx [hunk 9]
```diff
@@ -255,28 +260,28 @@ function MoneyTemporaryForRefactorRequestParticipantsSelector({participants, onF
...
-    const hasPolicyExpenseChatParticipant = lodashSome(participants, (participant) => participant.isPolicyExpenseChat);
+    const hasPolicyExpenseChatParticipant = participants.some((participant) => participant.isPolicyExpenseChat);
     const shouldShowSplitBillErrorMessage = participants.length > 1 && hasPolicyExpenseChatParticipant;
 
     // canUseP2PDistanceRequests is true if the iouType is track expense, but we don't want to allow splitting distance with track expense yet
     const isAllowedToSplit =
-        (canUseP2PDistanceRequests || iouRequestType !== CONST.IOU.REQUEST_TYPE.DISTANCE) &&
-        ![CONST.IOU.TYPE.PAY, CONST.IOU.TYPE.TRACK, CONST.IOU.TYPE.INVOICE].includes(iouType) &&
-        ![CONST.IOU.ACTION.SHARE, CONST.IOU.ACTION.SUBMIT, CONST.IOU.ACTION.CATEGORIZE].includes(action);
+        (canUseP2PDistanceRequests ?? iouRequestType !== CONST.IOU.REQUEST_TYPE.DISTANCE) &&
```
Comment (DylanDylann) [inline src/pages/iou/request/MoneyRequestParticipantsSelector.tsx:279]: Using ?? will change the behavior, could you help to explain about this change?

Comment (DylanDylann) [inline src/pages/iou/request/MoneyRequestParticipantsSelector.tsx:279]: Using ?? will change the behavior, could you help to explain about this change?

Comment (DylanDylann) [reply src/pages/iou/request/MoneyRequestParticipantsSelector.tsx:279]: It caused this bug: Can't make multiple selection  

https://github.com/Expensify/App/assets/141406735/14210648-6522-48f8-bb94-5342687f961a

Comment (pasyukevich) [reply src/pages/iou/request/MoneyRequestParticipantsSelector.tsx]: reverted 

There is no need for this change, missed logic during the migration

File: src/pages/iou/request/MoneyRequestParticipantsSelector.tsx (hunk 7 above)
Comment (pasyukevich) [reply src/pages/iou/request/MoneyRequestParticipantsSelector.tsx]: updated

File: src/pages/iou/request/MoneyRequestParticipantsSelector.tsx (hunk 8 above)
Comment (pasyukevich) [reply src/pages/iou/request/MoneyRequestParticipantsSelector.tsx]: updated to required

**Review by DylanDylann — APPROVED** [review]
LGTM 💯

**Review by chiragsalian — APPROVED** [review]
Code LGTM but we're on a merge freeze this week so I'm holding off merging at this time.

**Comment by tgolen** [comment]
I've removed the HOLD on this and will go ahead and merge it.

**Comment by tgolen** [comment]
Sure, can you please comment on https://github.com/Expensify/Expensify/issues/395017 and I'll assign it to you.

**Comment by DylanDylann** [comment]
@tgolen This is an internal repo. I can't access to it

**Comment by tgolen** [comment]
Arg, wrong repo. Sorry!
